Mathlib/Data/Nat/Cast/Order.lean

```diff
@@ -34,6 +34,9 @@ theorem mono_cast : Monotone (Nat.cast : ℕ → α) :=
     rw [Nat.cast_succ]; exact le_add_of_nonneg_right zero_le_one
 #align nat.mono_cast Nat.mono_cast
 
+@[gcongr]
+theorem cast_le_cast {a b : ℕ} (h : a ≤ b) : (a : α) ≤ b := mono_cast h
```

Comment:
```suggestion
theorem GCongr.cast_le_cast {a b : ℕ} (h : a ≤ b) : (a : α) ≤ b := mono_cast h
```

Reply:
Same comment as above.

Reply:
This is a lemma internal to `GCongr`. If we don't namespace it, people, will be confused between `Nat.cast_le`, `Nat.cast_le_cast` and `Nat.cast_mono`. Further, the argument that we don't have much in the `GCongr` namespace doesn't hold since `gcongr` is so new.

Reply:
This lemma was (not-namespaced) in `Tactic/GCongr`. We can move it to the `GCongr` namespace and deprecate the old lemma in favor of `Nat.mono_cast`.

Reply:
I would consider the original lack of namespacing as an accident. I think we should not even deprecate but delete, since `Nat.cast_le` should really be called `Nat.cast_le_cast` and the current `Nat.cast_le_cast` hasn't been around for long.

Reply:
I deprecated old lemmas.